Added support for some ES6 features

DonaldDuck313 opened the pull request:
Added support for let keyword, for(... of ...) loops, arrow functions, optional function parameters and classes. I will also suggest an edit to process.php to go with this one.

drify commented:
Oh!
I never expect someone would _actually_ use this project! Since it was too slow for using in my production site. Does it really help you? It's my pleasure if that's true.

drify commented:
I'm not very familiar with ES6. Do you have test cases?

DonaldDuck313 commented:
If you mean code you can test it on, you can use the following code for example, which tests all the features that I've added. Most of the code is from the MDN page about classes, and I've added some stuff to test other features.

```
class Polygon {
  constructor(height, width) {
    this.height = height;
    this.width = width;
  }
  
  get area() {
    return this.calcArea();
  }

  calcArea() {
    return this.height * this.width;
  }
}

let square = new Polygon(10, 10);

console.log(square.area);

class Point {
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }

  static distance(a, b) {
    const dx = a.x - b.x;
    const dy = a.y - b.y;

    return Math.sqrt(dx*dx + dy*dy);
  }
}

let p1 = new Point(5, 5);
let p2 = new Point(10, 10);

console.log(Point.distance(p1, p2));

class Animal { 
  constructor(name = "Anonymous animal") {
    this.name = name;
  }
  
  speak() {
    console.log(this.name + ' makes a noise.');
  }

  static eat() {
    return this;
  }
}

class Dog extends Animal {
  speak() {
    console.log(this.name + ' barks.');
  }
}

let d = new Dog('Mitzie');
d.speak();

//For testing arrow functions and for of loops
let makeDogsBark = () => {
    let lotsOfDogs = [
        new Dog('Rocket'),
        new Dog('Perri'),
        new Dog('Bonnie')
    ];
    for(let dog of lotsOfDogs){
        dog.speak();
    }
};

makeDogsBark();
```

It's true that it's slow. I will look into ways of optimizing it, and in the worst of cases I can always cache the result and only minify files that have changed more recently than the cached version. That would still be more useful than any other minifier since I won't have to think of minifying the files manually anymore.
